rest/server_context.go

```diff
@@ -815,21 +815,44 @@ func (sc *ServerContext) installPrincipals(context *db.DatabaseContext, spec map
 		} else {
 			princ.Name = &name
 		}
-		_, err := context.UpdatePrincipal(*princ, (what == "user"), isGuest)
-		if err != nil {
-			// A conflict error just means updatePrincipal didn't overwrite an existing user.
-			if status, _ := base.ErrorAsHTTPStatus(err); status != http.StatusConflict {
-				return err
+
+		worker := func() (shouldRetry bool, err error, value interface{}) {
+			_, err = context.UpdatePrincipal(*princ, (what == "user"), isGuest)
+			if err != nil {
+				if base.IsConflictError(err) {
+					// Ignore and absorb this error if it's a conflict error, which just means that updatePrincipal didn't overwrite an existing user.
+					// Since if there's an existing user it's "mission accomplished", this can be treated as a success case.
+					return false, nil, nil
+				}
+				// Assume it might be a timeout error due to view re-indexing.  So retry
```

Comment:
Should this be comparing against `ErrViewTimeoutError` and only retry in that case?

Reply:
Yeah - I think we should be handling the timeout error specifically, and not just retrying on everything except CAS.

Reply:
Fixed